packages/axelar-gmp-contract/src/evm-account-kit.js

```diff
@@ -0,0 +1,373 @@
+      return harden({
+        evmAccountAddress: /** @type {string | undefined} */ (undefined),
+        latestMessage:
+          /** @type {{ success: boolean; result: `0x${string}` }[] | undefined} */ (
+            undefined
+          ),
+        ...initialState,
+      });
+    },
+    {
+      tap: {
+        /**
+         * @param {VTransferIBCEvent} event
+         */
+        receiveUpcall(event) {
+          trace('receiveUpcall', event);
+
+          const tx = /** @type {FungibleTokenPacketData} */ (
+            JSON.parse(atob(event.packet.data))
+          );
+
+          trace('receiveUpcall packet data', tx);
+          /** @type {AxelarGmpIncomingMemo} */
+          const memo = JSON.parse(tx.memo);
+
+          if (memo.source_chain === 'Ethereum') {
+            const payloadBytes = decodeBase64(memo.payload);
+            const [{ isContractCallResult, data }] = decodeAbiParameters(
+              [
+                {
+                  type: 'tuple',
+                  components: [
+                    { name: 'isContractCallResult', type: 'bool' },
+                    {
+                      name: 'data',
+                      type: 'tuple[]',
+                      components: [
+                        { name: 'success', type: 'bool' },
+                        { name: 'result', type: 'bytes' },
+                      ],
+                    },
+                  ],
+                },
+              ],
+              payloadBytes,
+            );
+
+            trace(
+              'receiveUpcall Decoded:',
+              JSON.stringify({ isContractCallResult, data }),
+            );
+
+            if (this.state.evmAccountAddress) {
+              trace('Setting latestMessage:', data);
+              this.state.latestMessage = harden([...data]);
+            } else {
+              const [message] = data;
+              const { success, result } = message;
+
+              trace('Contract Call Status:', success);
+
+              if (success) {
+                const [address] = decodeAbiParameters(
+                  [{ type: 'address' }],
+                  result,
+                );
+                this.state.evmAccountAddress = address;
+                trace('evmAccountAddress:', this.state.evmAccountAddress);
+              }
+            }
+          }
+
+          trace('receiveUpcall completed');
+        },
+      },
+      transferWatcher: {
+        /**
+         * @param {void} _result
+         * @param {bigint} value the qty of uatom to delegate
+         */
+        onFulfilled(_result, value) {
+          trace('onFulfilled _result:', JSON.stringify(_result));
+          trace('onFulfilled value:', JSON.stringify(value));
+          trace('onFulfilled state:', JSON.stringify(this.state));
+        },
+      },
+      holder: {
+        getLocalAddress() {
+          return this.state.localAccount.getAddress().value;
+        },
+        async getAddress() {
```

Comment:
`getAddress` is ambiguous.

```suggestion
        async getEVMAddress() {
```

or `getRemoteAddress`